Embedded struct function calling causes invalid c pointer generation

OS: windows, Microsoft Windows 10 M~ v19042 64 줸
Processor: 16 cpus, 64bit, little endian, Intel(R) Core(TM) i7-10870H CPU @ 2.20GHz
CC version: Error: exec failed (CreateProcess) with code 2: 系統找不到指定的檔案。
 cmd: cc --version

getwd: C:\Users\User\v-jvm
vmodules: C:\Users\User\.vmodules
vroot: C:\Users\User\v
vexe: C:\Users\User\v\v.exe
vexe mtime: 2022-09-20 10:36:35
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.1 a9d63d7.bbf1ba4

Git version: git version 2.33.0.windows.2
Git vroot status: weekly.2022.38-2-gbbf1ba45
.git/config present: true
thirdparty/tcc status: thirdparty-windows-amd64 125e21e2

**What did you do?**
`v run main.v`

`main.v`:
```v
struct Access {}

fn (access &Access) acc() bool {
	return true
}

struct Field {
	Access
}

fn main() {
	mut fields := []&Field{}

	for mut field in fields {
		println(field.acc())
	}
}
```

**What did you expect to see?**
```
true
```
 
**What did you see instead?**
```cmd
==================
C:/Users/User/AppData/Local/Temp/v_0/test.15378175371196996901.tmp.c:7384: warning: cast between pointer and integer of different size
C:/Users/User/AppData/Local/Temp/v_0/test.15378175371196996901.tmp.c:12196: error: expected struct or union but not 'struct main__Field *'
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
```

## Workaround
```v
struct Access {}

fn (access &Access) acc() bool {
	return true
}

struct Field {
	Access
}

fn main() {
	mut fields := []&Field{}

	for mut field in fields {
		println(field.Access.acc())
	}
}
```

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1086,6 +1086,16 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 			g.write('(map[]){')
 			g.expr(node.left)
 			g.write('}[0]')
+		} else if node.from_embed_types.len > 0 {
+			n_ptr := node.left_type.nr_muls() - 1
+			if n_ptr > 0 {
+				g.write('(')
+				g.write('*'.repeat(n_ptr))
+				g.expr(node.left)
+				g.write(')')
+			} else {
+				g.expr(node.left)
+			}
 		} else {
 			g.expr(node.left)
 		}

--- a/vlib/v/tests/embed_method_call_test.v
+++ b/vlib/v/tests/embed_method_call_test.v
@@ -1,0 +1,23 @@
+struct Access {}
+
+fn (access &Access) acc() bool {
+	return true
+}
+
+struct Field {
+	Access
+}
+
+fn test_embed_method_call() {
+	mut fields := []&Field{}
+	fields << &Field{}
+
+	mut rets := []bool{}
+	for mut field in fields {
+		println(field.acc())
+		rets << field.acc()
+	}
+
+	assert rets.len == 1
+	assert rets[0]
+}

